pkg/api/api/account.go

```diff
@@ -4,12 +4,183 @@ import (
+	res, err := s.accountClient.CreateAccountV2(
+		ctx,
+		&accountproto.CreateAccountV2Request{
+			OrganizationId:   request.OrganizationId,
+			Email:            request.Email,
+			Name:             request.Name,
+			AvatarImageUrl:   request.AvatarImageUrl,
+			OrganizationRole: request.OrganizationRole,
+			EnvironmentRoles: request.EnvironmentRoles,
+			FirstName:        request.FirstName,
+			LastName:         request.LastName,
+			Language:         request.Language,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if res == nil {
+		s.logger.Error("Failed to create account: nil response",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.String("email", request.Email),
+				zap.String("organizationId", request.OrganizationId),
+				zap.String("role", request.OrganizationRole.String()),
+			)...,
+		)
+		return nil, ErrInternal
+	}
+
+	return &gwproto.CreateAccountV2Response{
+		Account: res.Account,
+	}, nil
+}
+
+func (s *grpcGatewayService) UpdateAccountV2(
+	ctx context.Context,
+	request *gwproto.UpdateAccountV2Request,
+) (*gwproto.UpdateAccountV2Response, error) {
+	envAPIKey, err := s.checkRequest(ctx, []accountproto.APIKey_Role{
+		accountproto.APIKey_PUBLIC_API_WRITE,
+		accountproto.APIKey_PUBLIC_API_ADMIN,
+	})
+	if err != nil {
+		s.logger.Error("Failed to check update account request",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.Error(err),
+				zap.String("email", request.Email),
+				zap.String("organizationId", request.OrganizationId),
+			)...,
+		)
+		return nil, err
+	}
+
+	headerMetaData := metadata.New(map[string]string{
+		role.APIKeyTokenMDKey:      envAPIKey.ApiKey.ApiKey,
+		role.APIKeyMaintainerMDKey: envAPIKey.ApiKey.Maintainer,
+		role.APIKeyNameMDKey:       envAPIKey.ApiKey.Name,
+	})
+	ctx = metadata.NewOutgoingContext(ctx, headerMetaData)
+
+	// delete account
+	if request.Deleted != nil && request.Deleted.Value {
+		_, err := s.accountClient.DeleteAccountV2(
+			ctx,
+			&accountproto.DeleteAccountV2Request{
+				Email:          request.Email,
+				OrganizationId: request.OrganizationId,
+			},
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		return &gwproto.UpdateAccountV2Response{}, nil
+	}
+
+	// disable account
+	if request.Disabled != nil && request.Disabled.Value {
+		_, err := s.accountClient.DisableAccountV2(
+			ctx,
+			&accountproto.DisableAccountV2Request{
+				Email:          request.Email,
+				OrganizationId: request.OrganizationId,
+			},
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		return &gwproto.UpdateAccountV2Response{}, nil
+	}
+
+	// enable account
+	if request.Disabled != nil && !request.Disabled.Value {
+		_, err := s.accountClient.EnableAccountV2(
+			ctx,
+			&accountproto.EnableAccountV2Request{
+				Email:          request.Email,
+				OrganizationId: request.OrganizationId,
+			},
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		return &gwproto.UpdateAccountV2Response{}, nil
+	}
```

Comment:
We need to use the Update API to enable and disable requests. It is already implemented in the domain.
https://github.com/bucketeer-io/bucketeer/blob/main/pkg/account/domain/account.go#L120-L122

Reply:
Oh I forgot that I was the one who implement it :)) I have exposed `disabled` field in web api and use it in gateway layer.